Library/Homebrew/test/dev-cmd/bottle_spec.rb

```diff
@@ -194,3 +194,104 @@ def stub_hash(parameters)
+            "name":           "testball",
+            "version":        "1.0",
+            "path":           "#{core_tap.path}/Formula/testball.rb",
+            "cellar":         "any_skip_relocation",
+            "os":             "big_sur",
+            "filename":       "hello-1.0.big_sur.bottle.tar.gz",
+            "local_filename": "hello--1.0.big_sur.bottle.tar.gz",
+            "sha256":         "a0af7dcbb5c83f6f3f7ecd507c2d352c1a018f894d51ad241ce8492fa598010f",
+          },
+        ),
+      )
+    end
+
+    File.open("testball-1.0.catalina.bottle.json", "w") do |f|
+      f.write(
+        stub_hash(
+          {
+            "name":           "testball",
+            "version":        "1.0",
+            "path":           "#{core_tap.path}/Formula/testball.rb",
+            "cellar":         "any_skip_relocation",
+            "os":             "catalina",
+            "filename":       "testball-1.0.catalina.bottle.tar.gz",
+            "local_filename": "testball--1.0.catalina.bottle.tar.gz",
+            "sha256":         "5334dd344986e46b2aa4f0471cac7b0914bd7de7cb890a34415771788d03f2ac",
+          },
+        ),
+      )
+    end
+
+    expect {
+      brew "bottle",
+           "--merge",
+           "--write",
+           "testball-1.0.big_sur.bottle.json",
+           "testball-1.0.catalina.bottle.json"
+    }.to output(
+      <<~EOS,
+        ==> testball
+          bottle do
+            root_url "#{HOMEBREW_BOTTLE_DEFAULT_DOMAIN}"
+            cellar :any_skip_relocation
+            sha256 "a0af7dcbb5c83f6f3f7ecd507c2d352c1a018f894d51ad241ce8492fa598010f" => :big_sur
+            sha256 "5334dd344986e46b2aa4f0471cac7b0914bd7de7cb890a34415771788d03f2ac" => :catalina
+          end
+      EOS
+    ).to_stdout
+
+    FileUtils.rm_f "rm testball-1.0.catalina.bottle.json"
+    FileUtils.rm_f "rm testball-1.0.big_sur.bottle.json"
+
+    expect((core_tap.path/"Formula/testball.rb").read).to eq(
+      <<~EOS,
+        class Testball < Formula
+          desc "Some test"
+        homepage "https://brew.sh/testball"
+        url "file://#{TEST_FIXTURE_DIR}/tarballs/testball-0.1-linux.tbz"
+        sha256 "fa7fac451a7c37e74f02e2a425a76aff89106098a55707832a02be5af5071cf8"
+
+          bottle do
+            root_url "#{HOMEBREW_BOTTLE_DEFAULT_DOMAIN}"
+            cellar :any_skip_relocation
+            sha256 "a0af7dcbb5c83f6f3f7ecd507c2d352c1a018f894d51ad241ce8492fa598010f" => :big_sur
+            sha256 "5334dd344986e46b2aa4f0471cac7b0914bd7de7cb890a34415771788d03f2ac" => :catalina
+          end
+
+        option "with-foo", "Build with foo"
+
+        def install
+          (prefix/"foo"/"test").write("test") if build.with? "foo"
+          prefix.install Dir["*"]
+          (buildpath/"test.c").write \
+            "#include <stdio.h>\\nint main(){printf(\\"test\\");return 0;}"
+          bin.mkpath
+          system ENV.cc, "test.c", "-o", bin/"test"
+        end
+
+
+
+        # something here
+
+        end
```

Comment:
Found out why the indentation here is weird: https://github.com/Homebrew/brew/blob/8fe91aa/Library/Homebrew/test/support/helper/spec/shared_context/integration_test.rb#L174 (it doesn't support multiline content)